test/uk/gov/hmrc/agentservicesaccount/controllers/SuspendedJourneyControllerSpec.scala

```diff
@@ -85,62 +103,141 @@ class SuspendedJourneyControllerSpec extends BaseISpec {
+        "email" -> "romel@romel.com",
+          "phone" -> "01711111111")
       val response = controller.submitContactDetails()(request)
 
       status(response) shouldBe SEE_OTHER
 
+      redirectLocation(await(response)) shouldBe Some(routes.SuspendedJourneyController.showSuspendedDescription().url)
+    }
+  }
+
+  "showSuspendedDescription" should {
+    "return Ok and show the description recovery page" in {
+      givenAuthorisedAsAgentWith(arn)
+      givenSuspensionStatus(SuspensionDetails(suspensionStatus = true, None))
+
+      expectGetSessionItemNone[String](DESCRIPTION)
 
+      val response: Future[Result] = controller.showSuspendedDescription()(fakeRequest())
+      status(response) shouldBe OK
+      Helpers.contentType(response).get shouldBe HTML
+
+      val content = Helpers.contentAsString(response)
+      content should include(messagesApi("suspend.description.title"))
+      content should include(messagesApi("suspend.description.hint1"))
+      content should include(messagesApi("suspend.description.h1"))
+      content should include(messagesApi("suspend.description.label"))
+      content should include(messagesApi("suspend.description.hint"))
+      content should include(messagesApi("common.continue-save"))
     }
   }
+
+
+  "showSuspendedSummary" should {
+    s"redirect to ${routes.SuspendedJourneyController.showContactDetails().url} when no details are present" in {
+      givenAuthorisedAsAgentWith(arn)
+      givenSuspensionStatus(SuspensionDetails(suspensionStatus = true, None))
+
+      expectGetSessionItemNone[String](NAME)
+      expectGetSessionItemNone[String](EMAIL)
+      expectGetSessionItemNone[String](PHONE)
+      expectGetSessionItemNone[String](DESCRIPTION)
+      expectGetSessionItemNone[String](ARN)
+
+      val response: Future[Result] = controller.showSuspendedSummary()(fakeRequest())
+      status(response) shouldBe 303
+      redirectLocation(await(response)) shouldBe Some(routes.SuspendedJourneyController.showContactDetails().url)
+    }
+    s"return Ok when session details are found" in {
+      givenAuthorisedAsAgentWith(arn)
+      givenSuspensionStatus(SuspensionDetails(suspensionStatus = true, None))
+
+      expectGetSessionItem[String](NAME, "Romel", 1)
+      expectGetSessionItem[String](EMAIL, "Romel@romel.com", 1)
+      expectGetSessionItem[String](PHONE, "017111111111", 1)
+      expectGetSessionItem[String](DESCRIPTION, "Some description", 1)
+      expectGetSessionItem[String](ARN, "XARN000001122", 1)
+
+
+      val response: Future[Result] = controller.showSuspendedSummary()(fakeRequest())
+      status(response) shouldBe OK
+    }
+
+  }
+
+
+  "submitSuspendedConfirmation" should {
+    "return Ok and show the confirmation page" in {
+      givenAuthorisedAsAgentWith(arn)
+      givenSuspensionStatus(SuspensionDetails(suspensionStatus = true, Some(Set("AGSV"))))
+
+      val response: Future[Result] = controller.showSuspendedConfirmation()(fakeRequest("POST", "/home"))
```

Comment:
this doesn't line up to the test description

Reply:
OK

Reply:
the http method is a GET here not POST

Reply:
you need tests for submitSuspendedSummary for 2 cases - details are in the session and details are not in the session.